src/org/labkey/test/components/glassLibrary/grids/DetailTableEdit.java

```diff
@@ -164,38 +165,24 @@ public boolean getBooleanField(String fieldCaption)
     public DetailTableEdit setBooleanField(String fieldCaption, boolean value)
     {
 
-        if(isFieldEditable(fieldCaption))
-        {
+        WebElement fieldValueElement = elementCache().fieldValue(fieldCaption).findElement(getComponentElement());
+        Assert.isTrue(isEditableField(fieldValueElement), String.format("Field '%s' is not editable, and cannot be set.", fieldCaption));
```

Comment:
This is from non-standard assert library (`org.eclipse.core.runtime.Assert`)
```suggestion
        Assert.assertTrue(String.format("Field '%s' is not editable, and cannot be set.", fieldCaption), isEditableField(fieldValueElement));
```

Reply:
D'oh! Fixed.